### PR TITLE
fix #44013: aliasing in property destructuring

### LINKED ISSUE #44013: Evaluation order in LHS struct destructuring
Consider:
```
julia> struct Foo
          x
          f
       end

julia> f = Foo(1, 2)
Foo(1, 2)
```
we have:
```
julia> (; x, f) = f
2
```
but
```
julia> (; f, x) = f
ERROR: type Int64 has no field x
Stacktrace:
 [1] getproperty(x::Int64, f::Symbol)
   @ Base ./Base.jl:38
 [2] top-level scope
   @ REPL[6]:1
```
Also in the first case, the overall return is the new value of `f`, but I believe it would be more consistent to return the old value of f. I think the syntax should work in phases, where it collects all the values into ssa values and only then applies the updates.

### PR DESCRIPTION
**Pull request by simeonschaub**

### CODE REVIEW

File: src/julia-syntax.scm [hunk 1]
```diff
@@ -2118,11 +2118,23 @@
...
+        ((and (pair? lhss) (vararg? (last lhss))
+              (eventually-call? (cadr (last lhss))))
+         (gensy))
+        (else (make-ssavalue))))
+
+(define (expand-property-destruct lhs x)
+  (if (not (length= lhs 1))
+      (error (string "invalid assignment location \"" (deparse `(tuple ,lhs)) "\"")))
+  (define (in-lhs? x lhss)
+    (any (lambda (l) (eq? l x)) lhss))
```
Comment (JeffBezanson) [inline src/julia-syntax.scm:2134]:
```suggestion
    (memq x lhss))
```

File: src/julia-syntax.scm [hunk 2]
```diff
@@ -2118,11 +2118,23 @@
...
+  (let* ((lhss (cdar lhs))
+         (xx   (maybe-ssavalue lhss x in-lhs?))
```
Comment (simeonschaub) [inline src/julia-syntax.scm:2136]:
```suggestion
  (let* ((lhss (cdar lhs))
         (xx   (maybe-ssavalue lhss x memq))
```

File: src/julia-syntax.scm (hunk 1 above)
Comment (simeonschaub) [reply src/julia-syntax.scm]: Don't know why I didn't think of that. Really no need to define an extra closure in that case.

File: src/julia-syntax.scm [hunk 3]
```diff
@@ -2118,11 +2118,21 @@
...
+             (ssavalue? x))
+          x)
+        ((and (pair? lhss) (vararg? (last lhss))
+              (eventually-call? (cadr (last lhss))))
+         (gensy))
+        (else (make-ssavalue))))
+
+(define (expand-property-destruct lhs x)
+  (if (not (length= lhs 1))
+      (error (string "invalid assignment location \"" (deparse `(tuple ,lhs)) "\"")))
```
Comment (JeffBezanson) [inline src/julia-syntax.scm:2132]: The other error below also needs the `(tuple ,lhs)` wrapping I think.

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**